### ISSUE TITLE
Jumping height in PAL

### ISSUE THREAD

**Issue by SkyTheLeafeon** [andwn/cave-story-md#270]
![cavestorygenprob7](https://user-images.githubusercontent.com/36712090/38007574-ee1747b6-31fe-11e8-8e2d-9d284f90f25a.gif)
I don't think this jump is supposed to be possible.

**Comment by SkyTheLeafeon** [comment]
Come to think of it, this affects everything else that jumps. Everything else jumps too high - it's not just Quote.

**Comment by andwn** [comment]
I'm not sure the reason. The speeds for player jumping are the same as Pixel used.

```
        if(pal_mode) {
		jumpSpeed = 0x500;
		gravity = 0x50;
		gravityJump = 0x20;
		maxFallSpeed = 0x5FF;
	} else {
		jumpSpeed = 0x42A;
		gravity = 0x42;
		gravityJump = 0x1A;
		maxFallSpeed = 0x4FF;
	}
	if(player.underwater) {
		jumpSpeed >>= 1;
		gravity >>= 1;
		gravityJump >>= 1;
		maxFallSpeed >>= 1;
	}
```

Guessing there is something in the logic, comparing to PC there is an extra pixel or two of height.

Every object uses unique numbers for speeds.

**Comment by andwn** [comment]
I think I am just going to decrease the jump speed very slightly, until I can no longer make that specific jump.

**Comment by andwn** [comment]
Apparently this is the "sweet spot" where you can't make the jump in your gif but can still hop over Booster.

    jumpSpeed = 	0x4E0; // 0x500
    gravity = 	0x50;
    gravityJump = 	0x21; // 0x20
    maxFallSpeed =	0x5FF;

Pixel's engine probably applies gravity once immediately whereas I wait a frame. No idea about why I had to increase the button holding gravity though.

Oh well, numbers are numbers, what matters is it feels right.

**Comment by SkyTheLeafeon** [comment]
Hmm. Will test when we get to 0.5.2/0.6.0.

**Comment by andwn** [comment]
This one was fixed in 0.6.0
